### PR TITLE
NSURLError CustomStringConvertable conformance

### PR DESCRIPTION
**Pull request by dlbuckley**
I've added conformance to CustomStringConvertable to NSURL. I want to make sure that this is the intended support before adding additional support throughout the framework or if we are just going to rely on overriding the getter for the description property?

This is less about getting API parity with Foundation and more about making it bridge better with Swift.

### CODE REVIEW

File: Foundation/NSURL.swift [hunk 1]
```diff
@@ -342,6 +342,14 @@ public class NSURL : NSObject, NSSecureCoding, NSCopying {
     override internal var _cfTypeID: CFTypeID {
         return CFURLGetTypeID()
     }
+    
+    override public var description: String {
+        if let absoluteString = self.absoluteString {
```
Comment (phausler) [inline Foundation/NSURL.swift]: does this handle relative urls? i.e. ones created via:
```swift
public convenience init(fileURLWithPath path: String, relativeToURL baseURL: NSURL?)
```

File: TestFoundation/TestNSURL.swift [hunk 2]
```diff
@@ -343,4 +344,11 @@ class TestNSURL : XCTestCase {
         */
     }
 
+    func test_description() {
+        // test with file that exists
+        let path = TestNSURL.gFileExistsPath
+        let url = NSURL(fileURLWithPath: path)
+        
+        XCTAssertEqual(path, url.description, "expected description does not match")
```
Comment (phausler) [inline TestFoundation/TestNSURL.swift]: on the darwin version of Foundation NSURLs come out for their description as "relativeString -- baseURL" if the base url is set, else they are presented as "relativeString -- relativeString"

**Comment by parkera** [comment]
Let's drop the conformance with `CustomStringConvertible` and just focus on adding the correct description method implementation. We should consider adoption of the CustomStringConvertible protocol at a more global scope than just NSURL (e.g. NSObject).

**Comment by phausler** [comment]
Just as a note: NSObject now conforms to CustomStringConvertible so descendant classes do not need to do anything but override description

**Comment by dlbuckley** [comment]
Good to know! I will make the amendments soon but I've been rammed this week so haven't had chance to sit down and do anything with it.

**Comment by phausler** [comment]
Going to close this one since it has not had any updates to the requested changes in a bit, feel free to re-open when ready.

**Pull request closed without merging**